CommunityToolShedMvc/Controllers/CommunityController.cs

```diff
@@ -141,19 +140,55 @@ public ActionResult Create(FormCollection collection)
  
         }
 
+        [HttpGet]
         public ActionResult Join()
         {
             List<Community> communities = new List<Community>();
             CustomPrincipal currentUser = (CustomPrincipal)User;
 
             communities = DatabaseHelper.Retrieve<Community>(@"
-                select c.CommunityName
+                select c.CommunityName, c.Id
                 from Community c
-                where c.OwnerId != @Id
+                left join PersonCommunity pc 
+                on pc.CommunityId = c.Id and pc.PersonId = @PersonId
+                where pc.Id is null 
             ",
-               new SqlParameter("@Id", currentUser.Person.Id));
+               new SqlParameter("@PersonId", currentUser.Person.Id));
 
             return View(communities);
         }
+
+        public ActionResult joinCommunity(int id)
```

Comment:
Any action method that changes data on the server should only accept POST requests. Given that, this method should be decorated with an `HttpPost` attribute.

Reply:
Also, capitalize the first letter of the method name: `JoinCommunity`.